tests/ip_link.py

```diff
@@ -17,14 +17,15 @@ def runTest(self):
         board.expect(prompt)
         if "ip: not found" not in board.before:
             up_interfaces = re.findall('\d: ([A-Za-z0-9-\.]+)[:@].*state UP ', board.before)
+            ether_interfaces = re.findall('\d: ([A-Za-z0-9-\.]+)[:@].*\n.*link/ether ', board.before)
+            assert up_interfaces == ether_interfaces
```

Comment:
That's not the case, if you see the example output of ip link show command on all possible interfaces available on a Ci40 board, there is a specific link type associated with ethernet and wireless interfaces i.e. "link/ether".  I think this will always remain constant.  Now, even if say 'lo' show UP state and ethernet (eth0) goes DOWN or UNKNOWN, then the lists up_interfaces and ether_interfaces won't match

    up_interfaces = ['lo', 'eth1', 'wlan0', 'wlan0-1']
    ether_interfaces = ['eth0', 'eth1', 'wlan0', 'wlan0-1']

I think this sufficiently checks for getting the expected result.